test/integration/targets/azure_rm_sqlserver_facts/tasks/main.yml

```diff
@@ -0,0 +1,37 @@
+- name: Prepare random number
+  set_fact:
+    rpfx: "{{ 1000 | random }}{{ resource_group | hash('md5') | truncate(7, True, '') }}"
+  run_once: yes
+
+- name: Create SQL Server
+  azure_rm_sqlserver:
+    resource_group: "{{ resource_group }}"
+    name: "sqlsrv{{ rpfx }}"
+    location: eastus
+    admin_username: mylogin
+    admin_password: Testpasswordxyz12!
+
+- name: Gather facts SQL Server
+  azure_rm_sqlserver_facts:
+    resource_group: "{{ resource_group }}"
+    server_name: "sqlsrv{{ rpfx }}"
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
```

Comment:
We want to test the actual facts and their values. While you don't have to do it in this PR we should potentially look at adding these further checks to the `azure_rm_sqlserver` module and use the facts module to check that the operations worked as expected.